File: Ent_UsAdmin/Migrar.vb
Imports System.ComponentModel
Imports System.Data.SqlClient
Imports Microsoft.Office.Interop

Public Class Migrar
    Public Archivo As String
    Public columna As String
    Public fila As String
    Private Sub Migrar_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub

    Private Sub txtArchivo_OnValueChanged(sender As Object, e As EventArgs) Handles txtArchivo.OnValueChanged

    End Sub

    Private Sub txtArchivo_KeyDown(sender As Object, e As KeyEventArgs) Handles txtArchivo.KeyDown
        If e.KeyCode = Keys.F2 Then
            Dim abrircarpeta As OpenFileDialog
            abrircarpeta = New OpenFileDialog
            abrircarpeta.Filter = "Archivo de Excel|*.xls;*.xlsx"
            If abrircarpeta.ShowDialog = Windows.Forms.DialogResult.OK Then
                txtArchivo.Text = abrircarpeta.FileName
            Else
                MessageBox.Show("No se seleccionó ningún archivo")
            End If
            ImportarExcel.ShowDialog()

        End If
    End Sub

    Private Sub BunifuThinButton21_Click(sender As Object, e As EventArgs) Handles BunifuThinButton21.Click
        Try
            Cargando.Show()
            Cargando.MonoFlat_Label1.Text = "Cargando datos"
            Dim intLoopCounter As Integer
            Dim contador1 As Integer
            Dim contador As Integer = 1
            Dim cantidadregistros As Integer = 1

            Dim fname As String
            Dim objXLApp As Excel.Application
            objXLApp = New Excel.Application
            fname = txtArchivo.Text
            With objXLApp
                .Workbooks.Open(fname)
                .Workbooks(1).Worksheets(1).Select()

                For intLoopCounter = fila To CInt(.ActiveSheet.Cells.SpecialCells(Excel.XlCellType.xlCellTypeLastCell).Row)
                    If Convert.ToString(.Range("" & columna & "" & intLoopCounter).Value) = "" Then
                        Exit For


                    End If

                    cantidadregistros += 1

                Next intLoopCounter

                For intLoopCounter = fila To cantidadregistros + fila - 2
                    'panelespera.Visible = True

                    '  BunifuCircleProgressbar1.MaxValue = cantidadregistros + fila - 2
                    Cargando.MonoFlat_Label1.Text = "" & contador & " registros de " & cantidadregistros - 1 & ""
                    '  BunifuCircleProgressbar1.Value = intLoopCounter
                    ' Sql = "insert into hotel(Nombre,telefono,email) values('" & .Range("" & Combonombre.Text & "" & intLoopCounter) & "','" & .Range("" & Combotelefono.Text & "" & intLoopCounter) & "','" & .Range("" & Comboemail.Text & "" & intLoopCounter) & "')"

                    dtimpuestos.Rows.Add(.Range("" & columna & "" & intLoopCounter).Value, .Range("" & columna & "" & intLoopCounter).Offset(0, 1).Value, .Range("" & columna & "" & intLoopCounter).Offset(0, 2).Value, .Range("" & columna & "" & intLoopCounter).Offset(0, 3).Value)

                    contador += 1
                Next intLoopCounter
                .Workbooks(1).Close(False)
                .Quit()
            End With

        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
        Cargando.Close()
    End Sub

    Private Sub BackgroundWorker1_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundWorker1.DoWork

        iniciarconexionempresa()
        Dim canregistros As Integer
        canregistros = dtimpuestos.Rows.Count
        Dim numero As Integer = 1

        For Each row As DataGridViewRow In dtimpuestos.Rows
                Cargando.MonoFlat_Label1.Text = numero & " registros de '" & canregistros & "'"
                Dim comandoCalendario As SqlCommand
                Dim consultaCalendario As String
                Dim readerCalendario As SqlDataReader
                consultaCalendario = "select FechaPago,Monto,interes,fecha,estado,abonado,pendiente,npago,convenio from pagosext where id_credito = '" & row.Cells(0).Value & "' order by npago asc"
                comandoCalendario = New SqlCommand
                comandoCalendario.Connection = conexionempresa
                comandoCalendario.CommandText = consultaCalendario
                readerCalendario = comandoCalendario.ExecuteReader
                If readerCalendario.HasRows Then
                    While readerCalendario.Read
                        Dim consultaCalendarioSac As String
                    Dim comandoCalendarioSac As SqlCommand
                    Dim fechapago, monto, interes, pendiente, abonado, npago, id_credito, estado, convenio, fechaupago As String
                    fechapago = readerCalendario("fechapago")
                    monto = readerCalendario("monto")
                    interes = readerCalendario("interes")
                    pendiente = readerCalendario("pendiente")
                    abonado = readerCalendario("abonado")
                    npago = readerCalendario("npago")
                    id_credito = row.Cells(2).Value
                    estado = readerCalendario("estado")
                    convenio = readerCalendario("convenio")
                    fechaupago = readerCalendario("fecha")

                    consultaCalendarioSac = "insert into calendarionormal values('" & fechapago & "','" & monto & "','" & interes & "','" & pendiente & "','" & abonado & "','" & readerCalendario("npago") & "','" & id_credito & "','" & estado & "','" & convenio & "','" & fechaupago & "')"
                    comandoCalendarioSac = New SqlCommand
                        comandoCalendarioSac.Connection = conexionempresa
                        comandoCalendarioSac.CommandText = consultaCalendarioSac
                        comandoCalendarioSac.ExecuteNonQuery()

                    End While

                End If
                Dim comandoActEstado As SqlCommand
                Dim consultaActEstado As String
                consultaActEstado = "update credito set estado = '" & row.Cells(3).Value & "' where id = '" & row.Cells(2).Value & "'"
                comandoActEstado = New SqlCommand
                comandoActEstado.Connection = conexionempresa
                comandoActEstado.CommandText = consultaActEstado
                comandoActEstado.ExecuteNonQuery()

                numero += 1

            Next


    End Sub

    Private Sub BunifuThinButton22_Click(sender As Object, e As EventArgs) Handles BunifuThinButton22.Click
        Cargando.Show()
        Cargando.MonoFlat_Label1.Text = "Cargando Información"
        BackgroundWorker1.RunWorkerAsync()

    End Sub

    Private Sub BackgroundWorker1_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundWorker1.RunWorkerCompleted
        Cargando.Close()

    End Sub

    Private Sub Panel1_Paint(sender As Object, e As PaintEventArgs) Handles Panel1.Paint

    End Sub

    Private Sub Panel1_MouseDown(sender As Object, e As MouseEventArgs) Handles Panel1.MouseDown
        If e.Button = MouseButtons.Left Then
            MoveForm(Me)
        End If
    End Sub
End Class

File: Ent_UsAdmin/Migrar.cs
using System;
using System.ComponentModel;
using System.Data.SqlClient;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class Migrar
    {
        public string Archivo;
        public string columna;
        public string fila;

        public Migrar()
        {
            InitializeComponent();
        }
        private void Migrar_Load(object sender, EventArgs e)
        {

        }

        private void txtArchivo_OnValueChanged(object sender, EventArgs e)
        {

        }

        private void txtArchivo_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.F2)
            {
                OpenFileDialog abrircarpeta;
                abrircarpeta = new OpenFileDialog();
                abrircarpeta.Filter = "Archivo de Excel|*.xls;*.xlsx";
                if (abrircarpeta.ShowDialog() == DialogResult.OK)
                {
                    txtArchivo.Text = abrircarpeta.FileName;
                }
                else
                {
                    MessageBox.Show("No se seleccionó ningún archivo");
                }
                My.MyProject.Forms.ImportarExcel.ShowDialog();

            }
        }

        private void BunifuThinButton21_Click(object sender, EventArgs e)
        {
            try
            {
                My.MyProject.Forms.Cargando.Show();
                My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Cargando datos";
                int intLoopCounter;
                int contador1;
                int contador = 1;
                int cantidadregistros = 1;

                string fname;
                Microsoft.Office.Interop.Excel.Application objXLApp;
                objXLApp = new Microsoft.Office.Interop.Excel.Application();
                fname = txtArchivo.Text;
                objXLApp.Workbooks.Open(fname);
                objXLApp.Workbooks[1].Worksheets[(object)1].Select();

                var loopTo = Conversions.ToInteger(objXLApp.ActiveSheet.Cells.SpecialCells(Microsoft.Office.Interop.Excel.XlCellType.xlCellTypeLastCell).Row);
                for (intLoopCounter = Conversions.ToInteger(fila); intLoopCounter <= loopTo; intLoopCounter++)
                {
                    if (string.IsNullOrEmpty(Convert.ToString(objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Value())))
                    {
                        break;


                    }

                    cantidadregistros += 1;

                }

                var loopTo1 = (int)Math.Round(cantidadregistros + Conversions.ToDouble(fila) - 2d);
                for (intLoopCounter = Conversions.ToInteger(fila); intLoopCounter <= loopTo1; intLoopCounter++)
                {
                    // panelespera.Visible = True

                    // BunifuCircleProgressbar1.MaxValue = cantidadregistros + fila - 2
                    My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "" + contador + " registros de " + (cantidadregistros - 1) + "";
                    // BunifuCircleProgressbar1.Value = intLoopCounter
                    // Sql = "insert into hotel(Nombre,telefono,email) values('" & .Range("" & Combonombre.Text & "" & intLoopCounter) & "','" & .Range("" & Combotelefono.Text & "" & intLoopCounter) & "','" & .Range("" & Comboemail.Text & "" & intLoopCounter) & "')"

                    dtimpuestos.Rows.Add(objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Value(), objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Offset(0, 1).get_Value(), objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Offset(0, 2).get_Value(), objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Offset(0, 3).get_Value());

                    contador += 1;
                }
                objXLApp.Workbooks[1].Close(false);
                objXLApp.Quit();
            }

            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            My.MyProject.Forms.Cargando.Close();
        }

        private void BackgroundWorker1_DoWork(object sender, DoWorkEventArgs e)
        {
            try
            {
                Module1.iniciarconexionempresa();
                int canregistros;
                canregistros = dtimpuestos.Rows.Count;
                int numero = 1;

                foreach (DataGridViewRow row in dtimpuestos.Rows)
                {
                    My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = numero + " registros de '" + canregistros + "'";
                    SqlCommand comandoCalendario;
                    string consultaCalendario;
                    SqlDataReader readerCalendario;
                    consultaCalendario = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject("select FechaPago,Monto,interes,fecha,estado,abonado,pendiente,npago,convenio from pagosext where id_credito = '", row.Cells[0].Value), "' order by npago asc"));
                    comandoCalendario = new SqlCommand();
                    comandoCalendario.Connection = Module1.conexionempresa;
                    comandoCalendario.CommandText = consultaCalendario;
                    readerCalendario = comandoCalendario.ExecuteReader();
                    if (readerCalendario.HasRows)
                    {
                        while (readerCalendario.Read())
                        {
                            string consultaCalendarioSac;
                            SqlCommand comandoCalendarioSac;
                            string fechapago, monto, interes, pendiente, abonado, npago, id_credito, estado, convenio, fechaupago;
                            fechapago = Conversions.ToString(readerCalendario["fechapago"]);
                            monto = Conversions.ToString(readerCalendario["monto"]);
                            interes = Conversions.ToString(readerCalendario["interes"]);
                            pendiente = Conversions.ToString(readerCalendario["pendiente"]);
                            abonado = Conversions.ToString(readerCalendario["abonado"]);
                            npago = Conversions.ToString(readerCalendario["npago"]);
                            id_credito = Conversions.ToString(row.Cells[2].Value);
                            estado = Conversions.ToString(readerCalendario["estado"]);
                            convenio = Conversions.ToString(readerCalendario["convenio"]);
                            fechaupago = Conversions.ToString(readerCalendario["fecha"]);

                            consultaCalendarioSac = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject("insert into calendarionormal values('" + fechapago + "','" + monto + "','" + interes + "','" + pendiente + "','" + abonado + "','", readerCalendario["npago"]), "','"), id_credito), "','"), estado), "','"), convenio), "','"), fechaupago), "')"));
                            comandoCalendarioSac = new SqlCommand();
                            comandoCalendarioSac.Connection = Module1.conexionempresa;
                            comandoCalendarioSac.CommandText = consultaCalendarioSac;
                            comandoCalendarioSac.ExecuteNonQuery();

                        }

                    }
                    SqlCommand comandoActEstado;
                    string consultaActEstado;
                    consultaActEstado = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject("update credito set estado = '", row.Cells[3].Value), "' where id = '"), row.Cells[2].Value), "'"));
                    comandoActEstado = new SqlCommand();
                    comandoActEstado.Connection = Module1.conexionempresa;
                    comandoActEstado.CommandText = consultaActEstado;
                    comandoActEstado.ExecuteNonQuery();

                    numero += 1;

                }
            }

            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }


        }

        private void BunifuThinButton22_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.Cargando.Show();
            My.MyProject.Forms.Cargando.MonoFlat_Label1.Text = "Cargando Información";
            BackgroundWorker1.RunWorkerAsync();

        }

        private void BackgroundWorker1_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            My.MyProject.Forms.Cargando.Close();

        }

        private void Panel1_Paint(object sender, PaintEventArgs e)
        {

        }

        private void Panel1_MouseDown(object sender, MouseEventArgs e)
        {
            if (e.Button == MouseButtons.Left)
            {
                Module_resize.MoveForm(this);
            }
        }
    }
}